tedge_config/src/models/flag.rs

```diff
@@ -0,0 +1,85 @@
+use std::convert::{TryFrom, TryInto};
+
+/// Represents a boolean type.
+///
+/// We need this newtype in order to implement `TryFrom<String>` and `TryInto<String>`.
+/// The config_key macro uses query_string() and update_string().
+/// Therefore, boolean needs to be converted from/to String.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Eq, PartialEq)]
+#[serde(transparent)]
+pub struct Flag(pub bool);
+
+#[derive(thiserror::Error, Debug)]
+#[error("Failed to parse flag: {input}. Supported values are: true, false")]
+pub struct InvalidFlag {
+    input: String,
+}
+
+impl TryFrom<String> for Flag {
+    type Error = InvalidFlag;
+
+    fn try_from(input: String) -> Result<Self, Self::Error> {
+        match input.as_str() {
+            "true" => Ok(Flag(true)),
+            "false" => Ok(Flag(false)),
+            _ => Err(InvalidFlag { input }),
+        }
+    }
+}
+
+impl TryInto<String> for Flag {
+    type Error = std::convert::Infallible;
+
+    fn try_into(self) -> Result<String, Self::Error> {
+        match self {
+            Flag(true) => Ok(String::from("true")),
+            Flag(false) => Ok(String::from("false")),
+        }
+    }
+}
+
+impl Into<bool> for Flag {
+    fn into(self) -> bool {
+        self.0
+    }
+}
```

Comment:
Preferably we implement `From` trait which would imply `Into`, probably something like:
```suggestion
impl From<Flag> for bool {
    fn from(value: Flag) -> Self {
        value.0
    }
}
```